lib/Sema/CSGen.cpp

```diff
@@ -1505,7 +1505,7 @@ namespace {
           assert(adjointDecl->isInstanceMember() || adjointDecl->isStatic() &&
                  "Expected adjoint to be an instance/static method");
           auto methodContext = CurDC->getInnermostMethodContext();
-          baseType = methodContext->getParameterList(0)->get(0)->getType();
+          baseType = methodContext->getParameters()->get(0)->getType();
```

Comment:
getParameterList(0) was the 'self' parameter list. I think you want `methodContext->getImplicitSelfDecl()->getType()`.

Reply:
This was done here: https://github.com/apple/swift/commit/652b33b51dea88ce3fa190b43db2adb55221b082.

@bgogul, was this overridden by rebase somehow?

Reply:
This was done here: https://github.com/apple/swift/commit/652b33b51dea88ce3fa190b43db2adb55221b082.

@bgogul, was this overridden by rebase somehow?

Reply:
This commit is included in the merge. If you change the diff view include changes from all commits, you will see this is fixed.